cmd/pdc/main.go

```diff
@@ -107,6 +124,54 @@ func main() {
 		os.Exit(1)
 	}
 
+	sshKeyFileEnvVar, ok := os.LookupEnv("GCLOUD_SSH_KEY_FILE")
+	if ok && (sshConfig.KeyFile == "") {
+		sshConfig.KeyFile = sshKeyFileEnvVar
+	}
+
+	// Using the same logic as the `ssh.Config` definition, clamping to max of 3
+	sshLogLevelSelectEnvVar, envVarOk := os.LookupEnv("GCLOUD_SSH_LOG_LEVEL")
+	parsedSSHLogLevelSelectEnvVar, parseIntErr := strconv.Atoi(sshLogLevelSelectEnvVar)
+	if parseIntErr == nil && envVarOk && (sshConfig.LogLevel > 3 || sshConfig.LogLevel < 0) {
+		sshConfig.LogLevel = parsedSSHLogLevelSelectEnvVar
+	}
+
+	skipSSHValidationEnvVar, envVarOk := os.LookupEnv("GCLOUD_SSH_SKIP_SSH_VALIDATION")
+	parsedSSHValidationEnvVar, parseBoolErr := strconv.ParseBool(skipSSHValidationEnvVar)
+	if parseBoolErr == nil && envVarOk && !sshConfig.SkipSSHValidation {
+		sshConfig.SkipSSHValidation = parsedSSHValidationEnvVar
+	}
+
+	// We're actually going to append the values provided here to the already-parsed SSHFlags array
+	// As the argument can be provided more than once, this feels like the most compatible option
+	additionalSSHFlagsEnvVar, ok := os.LookupEnv("GCLOUD_SSH_ADDITIONAL_SSHFLAGS")
+	if ok {
+		sshConfig.SSHFlags = append(sshConfig.SSHFlags, strings.Split(additionalSSHFlagsEnvVar, " ")...)
+	}
+
+	sshForceKeyFileOverwriteEnvVar, envVarOk := os.LookupEnv("GCLOUD_SSH_FORCE_KEY_FILE_OVERWRITE")
+	parsedSSHForceKeyFileOverwriteEnvVar, parseBoolErr := strconv.ParseBool(sshForceKeyFileOverwriteEnvVar)
+	if parseBoolErr == nil && envVarOk && !sshConfig.ForceKeyFileOverwrite {
+		sshConfig.ForceKeyFileOverwrite = parsedSSHForceKeyFileOverwriteEnvVar
+	}
+
+	sshCertExpiryWindowEnvVar, envVarOk := os.LookupEnv("GCLOUD_SSH_CERT_EXPIRY_WINDOW")
+	parsedSSHCertExpiryWindowEnvVar, parseTimeErr := time.ParseDuration(sshCertExpiryWindowEnvVar)
+	if parseTimeErr == nil && envVarOk && (sshConfig.CertExpiryWindow == 0) {
```

Comment:
default here is 5 minutes - you'll need to compare against the flag's default value.

Reply:
I'd recommend moving these checks into some function in the ssh pkg. That way, we can set some private variable for the default duration for example, and use it in the flag definition and in here.

Reply:
Not quite sure what you mean here. Are you proposing that I approach this similarly to the `*cfg.ApplyEnvironment` structuring from #107?

Reply:
Yes, that looks like a good idea!

Reply:
Done in 111c4c751cf015056664b49076a43745df9abd81 - let me know what you think of the implementation. I couldn't figure out a good way to pass around the prefix without just making it a parameter to the `config.ApplyEnvironment` function, I don't know enough go to do anything trickier.